src/emqx_channel.erl

```diff
@@ -1536,6 +1536,7 @@ ensure_connected(Channel = #channel{conninfo = ConnInfo,
                                     clientinfo = ClientInfo}) ->
     NConnInfo = ConnInfo#{connected_at => erlang:system_time(millisecond)},
     ok = run_hooks('client.connected', [ClientInfo, NConnInfo]),
+    emqx_cm:increment_connected_client_count(),
```

Comment:
Should we register this in the hook above? :thinking:

Reply:
I'm afraid doing such things in a hook callback will make things unnecessary more complicated.
because we'd basically have to create a new plugin-like process if not application to manage the full lifecycle of hook -> unhook.